perfkitbenchmarker/traces/perfspect.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+PerfSpect is a system performance profiling and processing tool based on linux perf.
+
+Usage:
+  Required flags:
+  --perfspect
+
+  Example:
+  ./pkb.py --cloud=AWS --benchmarks=your_wl --machine_type=m5.2xlarge --os_type=ubuntu2004 --perfspect
+
+  Refer to ./perfkitbenchmarker/data/perfspect/README.md for more details on flags and usage
+"""
+
+import logging
+import os
+import posixpath
+
+from absl import flags
+from six.moves.urllib.parse import urlparse
+
+from perfkitbenchmarker import events
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker import data
+from perfkitbenchmarker import os_types
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_boolean('perfspect', False,
+                     'Install and run perfspect on the target system.')
+flags.DEFINE_string('perfspect_tarball', None,
+                    'Local path to perfspect tarball.')
+flags.DEFINE_string('perfspect_url', None,
+                    'URL for downloading perfspect tarball.')
+
+PERFSPECT_ARCHIVE_URL = "https://github.com/intel/PerfSpect/releases/download/v1.1.0/perfspect_1.1.0.tgz"
```

Comment:
use single quote here, please be consistent throughout the file.

in general, we only use double-quote with multi-line comments